File: slack-reader/src/main/java/dev/number6/slackreader/SnsService.java
package dev.number6.slackreader;

import dev.number6.slackreader.model.WorkspaceMessages;
import dev.number6.slackreader.port.NotificationPort;
import dev.number6.message.ChannelMessages;

public class SnsService {

    private final NotificationPort notifications;

    public SnsService(NotificationPort notifications) {
        this.notifications = notifications;
    }

    public void broadcastWorkspaceMessagesForActiveChannels(WorkspaceMessages messages) {
        messages.getActiveChannelNames()
                .stream()
                .map(m -> new ChannelMessages(m, messages.getMessagesForChannel(m), messages.getComprehensionDate()))
                .forEach(notifications::broadcast);
    }
}

File: slack-reader/src/main/kotlin/dev/number6/slackreader/SnsService.kt
package dev.number6.slackreader

import dev.number6.message.ChannelMessages
import dev.number6.slackreader.model.WorkspaceMessages
import dev.number6.slackreader.port.NotificationPort

class SnsService(private val notifications: NotificationPort) {
    fun broadcastWorkspaceMessagesForActiveChannels(messages: WorkspaceMessages) {
        messages.getActiveChannelNames()
                .map { m -> ChannelMessages(m, messages.getMessagesForChannel(m), messages.comprehensionDate) }
                .forEach { cm -> notifications.broadcast(cm) }
    }
}